### PR TITLE
BFD-3676: Add admit and principal diagnosis codes to Claim resource

### PR DESCRIPTION
**Pull request by aschey-forpeople**
**JIRA Ticket:**
BFD-3676

### What Does This PR Do?

- Adds the admit diagnosis code and the principal diagnosis code to the Claim resource
  - If any of the existing diagnosis codes are the same, they will have their claim type updated in place instead of adding a duplicate entry
  - If none of the existing codes match, new entries will be added.
- Updates the SAMHSA matcher tests to also check for scenarios where only the admit or principal diagnosis code contains SAMHSA entries.
- Updates the JSON endpoint responses to reflect the new fields.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [x] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

Have you fully verified and tested these changes? Is the acceptance criteria met? Please provide reproducible testing instructions, code snippets, or screenshots as applicable.

Unit and integration tests

Example diff between the API in `test` and the new version for a claim that has different values for principal and admit codes.

```diff
--- read1.json	2024-09-27 09:05:56
+++ read2.json	2024-09-27 09:06:02
@@ -1,957 +1,1001 @@
-// 20240927090523
-// https://test.bfd.cms.gov/v2/fhir/Claim/f-MjAwMDA0MDA0MTUxQkhDTFAgICAgICAxMjM0NQ?_format=json
+// 20240927090541
+// https://localhost:6500/v2/fhir/Claim/f-MjAwMDA0MDA0MTUxQkhDTFAgICAgICAxMjM0NQ?_format=json
 
 {
   "resourceType": "Claim",
   "id": "f-MjAwMDA0MDA0MTUxQkhDTFAgICAgICAxMjM0NQ",
   "meta": {
-    "lastUpdated": "2024-09-27T11:28:41.719-04:00"
+    "lastUpdated": "2024-09-27T08:28:41.719-07:00"
   },
   "contained": [
     {
       "resourceType": "Patient",
       "id": "patient",
       "identifier": [
         {
           "type": {
             "coding": [
               {
                 "system": "http://terminology.hl7.org/CodeSystem/v2-0203",
                 "code": "MC",
                 "display": "Patient's Medicare Number"
               }
             ]
           },
           "system": "http://hl7.org/fhir/sid/us-mbi",
           "value": "R0S2TEV9P5Y"
         }
       ],
       "name": [
         {
           "text": "s. Hobbes ([middle initial], [max 15 chars of last])",
           "family": "Hobbes",
           "given": [
             "s."
           ]
         }
       ],
       "birthDate": "2003-11-01"
     },
     {
       "resourceType": "Organization",
       "id": "provider-org",
       "identifier": [
         {
           "type": {
             "coding": [
               {
                 "system": "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBIdentifierType",
                 "code": "npi",
                 "display": "National Provider Identifier"
               }
             ]
           },
           "system": "http://hl7.org/fhir/sid/us-npi",
           "value": "0004971116"
         }
       ]
     }
   ],
   "extension": [
     {
       "url": "https://bluebutton.cms.gov/resources/variables/fiss/serv-typ-cd",
       "valueCoding": {
         "system": "https://bluebutton.cms.gov/resources/variables/fiss/serv-typ-cd",
         "code": "7"
       }
     }
   ],
   "identifier": [
     {
       "type": {
         "coding": [
           {
             "system": "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBIdentifierType",
             "code": "uc",
             "display": "Unique Claim ID"
           }
         ]
       },
       "system": "https://bluebutton.cms.gov/resources/variables/fiss/dcn",
       "value": "200004004151BHCLP      "
     }
   ],
   "status": "active",
   "type": {
     "coding": [
       {
         "system": "http://terminology.hl7.org/CodeSystem/claim-type",
         "code": "institutional",
         "display": "Institutional"
       }
     ]
   },
   "use": "claim",
   "patient": {
     "reference": "#patient"
   },
   "billablePeriod": {
     "start": "2023-11-10",
     "end": "2023-12-01"
   },
-  "created": "2024-09-27T12:05:23-04:00",
+  "created": "2024-09-27T09:05:41-07:00",
   "provider": {
     "reference": "#provider-org"
   },
   "priority": {
     "coding": [
       {
         "system": "http://terminology.hl7.org/CodeSystem/processpriority",
         "code": "normal",
         "display": "Normal"
       }
     ]
   },
   "facility": {
     "extension": [
       {
         "url": "https://bluebutton.cms.gov/resources/variables/fiss/lob-cd",
         "valueCoding": {
           "system": "https://bluebutton.cms.gov/resources/variables/fiss/lob-cd",
           "code": "8"
         }
       }
     ]
   },
   "diagnosis": [
     {
       "sequence": 1,
       "diagnosisCodeableConcept": {
         "coding": [
           {
             "system": "http://hl7.org/fhir/sid/icd-10-cm",
             "code": "w5p1m0d"
           }
         ]
       },
       "onAdmission": {
         "coding": [
           {
             "system": "https://bluebutton.cms.gov/resources/variables/fiss/diag-poa-ind",
             "code": "y"
           }
         ]
       }
     },
     {
       "sequence": 2,
       "diagnosisCodeableConcept": {
         "coding": [
           {
             "system": "http://hl7.org/fhir/sid/icd-10-cm",
             "code": "a6a3d4w"
           }
         ]
       },
       "onAdmission": {
         "coding": [
           {
             "system": "https://bluebutton.cms.gov/resources/variables/fiss/diag-poa-ind",
             "code": "a"
           }
         ]
       }
+    },
+    {
+      "sequence": 3,
+      "diagnosisCodeableConcept": {
+        "coding": [
+          {
+            "system": "http://hl7.org/fhir/sid/icd-10-cm",
+            "code": "c7l4s8h"
+          }
+        ]
+      },
+      "type": [
+        {
+          "coding": [
+            {
+              "system": "http://terminology.hl7.org/CodeSystem/ex-diagnosistype",
+              "code": "admitting",
+              "display": "Admitting Diagnosis"
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "sequence": 4,
+      "diagnosisCodeableConcept": {
+        "coding": [
+          {
+            "system": "http://hl7.org/fhir/sid/icd-10-cm",
+            "code": "k1n2k6e"
+          }
+        ]
+      },
+      "type": [
+        {
+          "coding": [
+            {
+              "system": "http://terminology.hl7.org/CodeSystem/ex-diagnosistype",
+              "code": "principal",
+              "display": "Principal Diagnosis"
+            }
+          ]
+        }
+      ]
     }
   ],
   "procedure": [
     {
       "sequence": 1,
-      "date": "2023-11-25T00:00:00-05:00",
+      "date": "2023-11-25T00:00:00-08:00",
       "procedureCodeableConcept": {
         "coding": [
           {
             "system": "http://www.cms.gov/Medicare/Coding/ICD10",
             "code": "brosbvv"
           }
         ]
       }
     },
     {
       "sequence": 2,
-      "date": "2023-11-15T00:00:00-05:00",
+      "date": "2023-11-15T00:00:00-08:00",
       "procedureCodeableConcept": {
         "coding": [
           {
             "system": "http://www.cms.gov/Medicare/Coding/ICD10",
             "code": "tqkozqf"
           }
         ]
       }
     },
     {
       "sequence": 3,
-      "date": "2023-11-23T00:00:00-05:00",
+      "date": "2023-11-23T00:00:00-08:00",
       "procedureCodeableConcept": {
         "coding": [
           {
             "system": "http://www.cms.gov/Medicare/Coding/ICD10",
             "code": "joboqxj"
           }
         ]
       }
     }
   ],
   "insurance": [
     {
       "sequence": 1,
       "focal": false,
       "coverage": {
         "identifier": {
           "system": "https://bluebutton.cms.gov/resources/variables/fiss/payers-name",
           "value": "Bubbles"
         }
       }
     },
     {
       "sequence": 2,
       "focal": false,
       "coverage": {
         "identifier": {
           "system": "https://bluebutton.cms.gov/resources/variables/fiss/payers-name",
           "value": "Kilroy"
         }
       }
     }
   ],
   "item": [
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 982308699
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "v"
               }
             ]
           }
         }
       ],
       "sequence": 1,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "7148"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
             "system": "https://bluebutton.cms.gov/resources/variables/fiss/rev-cd",
             "code": "q"
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "91228"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
-            "system": "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system": "https://www.cms.gov/Medicare/Coding/HCPCSReleaseCodeSets",
             "code": "c8w8h"
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "1",
               "code": "z"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "3",
               "code": "z"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "4",
               "code": "g"
             }
           ]
         }
       ],
       "servicedDate": "2023-10-29",
       "quantity": {
         "value": 827037445
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 26730069
           }
         }
       ],
       "sequence": 2,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "6439"
                 }
               ]
             }
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "26798"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
-            "system": "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system": "https://www.cms.gov/Medicare/Coding/HCPCSReleaseCodeSets",
             "code": "t2e1x"
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "1",
               "code": "q"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "2",
               "code": "v"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "3",
               "code": "u"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "4",
               "code": "u"
             }
           ]
         }
       ],
       "servicedDate": "2023-09-04",
       "quantity": {
         "value": 818479225
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 843605136
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "l"
               }
             ]
           }
         }
       ],
       "sequence": 3,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "7494"
                 }
               ]
             }
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "85178"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
-            "system": "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system": "https://www.cms.gov/Medicare/Coding/HCPCSReleaseCodeSets",
             "code": "g9o0g"
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "3",
               "code": "j"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "4",
               "code": "k"
             }
           ]
         }
       ],
       "servicedDate": "2023-11-06",
       "quantity": {
         "value": 690917875
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 869274566
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "i"
               }
             ]
           }
         }
       ],
       "sequence": 4,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "7792"
                 }
               ]
             }
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "40813"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
-            "system": "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system": "https://www.cms.gov/Medicare/Coding/HCPCSReleaseCodeSets",
             "code": "s6s1h"
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "1",
               "code": "n"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "2",
               "code": "i"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "3",
               "code": "y"
             }
           ]
         }
       ],
       "servicedDate": "2023-11-05",
       "quantity": {
         "value": 336540373
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 602730392
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "f"
               }
             ]
           }
         }
       ],
       "sequence": 5,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "6184"
                 }
               ]
             }
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "30193"
                 }
               ]
             }
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "1",
               "code": "e"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "2",
               "code": "s"
             }
           ]
         }
       ],
       "servicedDate": "2023-10-20",
       "quantity": {
         "value": 614731374
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 891033589
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "q"
               }
             ]
           }
         }
       ],
       "sequence": 6,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "0380"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
             "system": "https://bluebutton.cms.gov/resources/variables/fiss/rev-cd",
             "code": "y"
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "80780"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
-            "system": "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system": "https://www.cms.gov/Medicare/Coding/HCPCSReleaseCodeSets",
             "code": "i0k7y"
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "3",
               "code": "q"
             }
           ]
         }
       ],
       "servicedDate": "2023-09-01",
       "quantity": {
         "value": 545326567
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 884652353
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "y"
               }
             ]
           }
         }
       ],
       "sequence": 7,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "9047"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
             "system": "https://bluebutton.cms.gov/resources/variables/fiss/rev-cd",
             "code": "b"
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "04338"
                 }
               ]
             }
           }
         ],
         "coding": [
           {
-            "system": "https://bluebutton.cms.gov/resources/variables/hcpcs_cd",
+            "system": "https://www.cms.gov/Medicare/Coding/HCPCSReleaseCodeSets",
             "code": "q8n0u"
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "1",
               "code": "y"
             }
           ]
         },
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "4",
               "code": "q"
             }
           ]
         }
       ],
       "servicedDate": "2023-11-24",
       "quantity": {
         "value": 668447382
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 110321791
           }
         }
       ],
       "sequence": 8,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "9213"
                 }
               ]
             }
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "50424"
                 }
               ]
             }
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "2",
               "code": "f"
             }
           ]
         }
       ],
       "servicedDate": "2023-11-17",
       "quantity": {
         "value": 413803596
       }
     },
     {
       "extension": [
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/rev-serv-unit-cnt",
           "valueQuantity": {
             "value": 562727810
           }
         },
         {
           "url": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
           "valueCodeableConcept": {
             "coding": [
               {
                 "system": "https://bluebutton.cms.gov/resources/variables/fiss/hcpc-ind",
                 "code": "w"
               }
             ]
           }
         }
       ],
       "sequence": 9,
       "revenue": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/non-bill-rev-code",
                   "code": "9142"
                 }
               ]
             }
           }
         ]
       },
       "productOrService": {
         "extension": [
           {
             "url": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
             "valueCodeableConcept": {
               "coding": [
                 {
                   "system": "https://bluebutton.cms.gov/resources/variables/fiss/apc-hcpcs-apc",
                   "code": "29365"
                 }
               ]
             }
           }
         ]
       },
       "modifier": [
         {
           "coding": [
             {
               "system": "https://bluebutton.cms.gov/resources/codesystem/hcpcs",
               "version": "3",
               "code": "t"
             }
           ]
         }
       ],
       "servicedDate": "2023-10-22",
       "quantity": {
         "value": 191358331
       }
     }
   ],
   "total": {
     "value": 93452.36,
     "currency": "USD"
   }
 }

```

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/claimFissRead.json [hunk 1]
```diff
@@ -182,29 +182,29 @@
       }
     },
     {
+      "sequence": 3,
       "category": {
         "coding": [
           {
+            "system": "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBSupportingInfoType",
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v2/claimFissRead.json:189]: As we saw previously, it appears these files have been edited by hand and have inconsistent formatting. Most of the changes in these json files is fixing these inconsistencies.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/FissClaimTransformerV2.java [hunk 2]
```diff
@@ -268,37 +269,89 @@ private Reference getFacility(RdaFissClaim claimGroup) {
...
+    Stream<FissDiagnosisAdapterV2> extraCodes =
+        Stream.of(
+            new FissDiagnosisAdapterV2(claimGroup.getAdmitDiagCode()),
+            new FissDiagnosisAdapterV2(claimGroup.getPrincipleDiag()));
+
+    // Most of the time, the principal and admit codes should already be included in the list
+    // returned by getDiagCodes(),
+    // but if they're not, we need to make sure they're included in the final result.
+    // Here we'll concatenate them all together and filter out any duplicates.
+    return Stream.concat(diagnosisCodes, extraCodes).distinct();
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/FissClaimTransformerV2.java:313]: One place where this _could_ be different than the current behavior is if there are duplicate codes in the main diagnosis code list. Currently those would just be returned as is, but here the duplicates would be removed. I don't see why anyone would actually want to see the same code returned twice though so I don't think this is an issue.

**Review by dondevun — APPROVED** [review]
I think this is in good shape.

**Review by meliGuzman — APPROVED** [review]

**Review by alex-dzeda — APPROVED** [review]
Approved; checked the logical soundness of the new code, and checked that the SAMHSA codes used in tests are indeed ones that are in our list of codes. There's probably an opportunity with the rework to pull from the consolidated list in the future, instead of hard coding values.

**Pull request merged**